### PR TITLE
Add recipe for gpustat

### PR DESCRIPTION
**Pull request by hadim**
Require: https://github.com/conda-forge/staged-recipes/pull/8036

### CODE REVIEW

**Comment by hadim** [comment]
Any idea how to fix this issue?

```bash
      File "$PREFIX/lib/python3.7/site-packages/pkg_resources/__init__.py", line 782, in resolve
        replace_conflicting=replace_conflicting
      File "$PREFIX/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1065, in best_match
        return self.obtain(req, installer)
      File "$PREFIX/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1077, in obtain
        return installer(requirement)
      File "$PREFIX/lib/python3.7/site-packages/setuptools/dist.py", line 791, in fetch_build_egg
        return cmd.easy_install(req)
      File "$PREFIX/lib/python3.7/site-packages/setuptools/command/easy_install.py", line 667, in easy_install
        not self.always_copy, self.local_index
      File "$PREFIX/lib/python3.7/site-packages/setuptools/package_index.py", line 658, in fetch_distribution
        self.find_packages(requirement)
      File "$PREFIX/lib/python3.7/site-packages/setuptools/package_index.py", line 491, in find_packages
        self.scan_url(self.index_url + requirement.unsafe_name + '/')
      File "$PREFIX/lib/python3.7/site-packages/setuptools/package_index.py", line 831, in scan_url
        self.process_url(url, True)
      File "$PREFIX/lib/python3.7/site-packages/setuptools/package_index.py", line 326, in process_url
        "Be sure to add all dependencies in the meta.yaml  url=%s" % url)
    RuntimeError: Setuptools downloading is disabled in conda build. Be sure to add all dependencies in the meta.yaml  url=https://pypi.org/simple/pytest-runner/
Cleaning up...
  Removing source in /tmp/pip-req-build-ofxhvkvf
Removed file://$SRC_DIR from build tracker '/tmp/pip-req-tracker-igs_z8w4'
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-req-build-ofxhvkvf/
Removed build tracker '/tmp/pip-req-tracker-igs_z8w4'
Exception information:
Traceback (most recent call last):
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/cli/base_command.py", line 179, in main
    status = self.run(options, args)
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/commands/install.py", line 315, in run
    resolver.resolve(requirement_set)
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/resolve.py", line 131, in resolve
    self._resolve_one(requirement_set, req)
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/resolve.py", line 294, in _resolve_one
    abstract_dist = self._get_abstract_dist_for(req_to_install)
  File "$PREFIX/lib/python3.7/site-packages/pip/_internal/resolve.py", line 242, in _get_abstract_dist_
```

Ping @conda-forge/staged-recipes

**Comment by hadim** [comment]
See https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=24104

**Comment by chrisburr** [comment]
I think `pytest-runner` needs to be a host dependency

**Comment by hadim** [comment]
Thanks @chrisburr . Good to merge I think.

File: recipes/gpustat/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    - gpustat = gpustat:main
+
+requirements:
+  build:
+    - setuptools
+  host:
+    - python
+    - pip
+    - pytest-runner
+    - nvidia-ml
```
Comment (chrisburr) [inline recipes/gpustat/meta.yaml]: Are these all needed? The host section only needs the dependencies required for running the setup.py file.

File: recipes/gpustat/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 2f43421dbcd9ebd8caf179aeb5f78ac123786033fb9a4310ce0f8e18b25eb03e
+
+build:
+  skip: True  # [win]
```
Comment (chrisburr) [inline recipes/gpustat/meta.yaml]: Could this be a noarch package? [See here](http://conda-forge.org/docs/maintainer/knowledge_base.html#noarch%20builds)

It’s okay if windows then fails due to missing dependencies as the feedstock will only be built in Linux.

File: recipes/gpustat/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+  entry_points:
+    - gpustat = gpustat:main
+
+requirements:
+  build:
+    - setuptools
```
Comment (chrisburr) [inline recipes/gpustat/meta.yaml]: I think this belongs in the host section?

File: recipes/gpustat/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  run:
+    - python
+    - nvidia-ml
+    - psutil
+    - blessings >=1.6
+    - six >=1.7
+
+test:
+  imports:
+    - gpustat
```
Comment (chrisburr) [inline recipes/gpustat/meta.yaml]: Is there a command you could also test?

```
  commands:
    - gpustat --help
```

**Comment by hadim** [comment]
@chrisburr All our comments should have been addressed.

**Pull request merged**